caffe2/python/onnx/backend.py

```diff
@@ -294,42 +296,13 @@ def tensor2list(onnx_tensor):
 
     @classmethod
     def _rnn_shape_inference(cls, init_model, pred_model, n, input_blob, W):
-        # ad-hoc, informally-specified, bug-ridden, slow
-        # implementation of shape inference
-
-        # if the weight matrices are directly provided as
-        # initializers, their dimensions should be available in the
-        # init net model.
-        for x in init_model.graph.input:
+        for x in itertools.chain(init_model.graph.input,
+                                 init_model.graph.value_info,
+                                 pred_model.graph.input,
+                                 pred_model.graph.value_info):
             if x.name == W:
                 return x.type.tensor_type.shape.dim[1].dim_value
```

Comment:
check that it's non-empty and produce better error message?

Reply:
++